added some logic to make less error prone

ff05 opened the pull request:
I added the variable 'proxmox_new_vmid' to have the opportunity to give a fixed vmid. If the vmid is already in use, Proxmox will not create the VM

resize disk only worked when the primary disk was virtio0. Now the role detects correct disk.

In Debian Bookworm, the dhclient now uses the machine-id to request an IP-Address. But cloudinit does not set this dynamically.
Because of that, every created Machine is handled as the same by a DHCP Server. Now the Machine-ID will be recreated when this role is executed.

ff05 committed:
added variable to manually set new vmid

ff05 committed:
made primary_disk as variable if it is not virtio0

ff05 committed:
ensure machine-id is recreated for dhcp client

h3po reviewed (commented):
please check if it would be sufficient to remove the machine-id file from the template instead

h3po reviewed (commented):
please don't use proxmox_cloudimage_template_node, that is just a variable from the example code and does not exist within the role. use $(hostname -s) if it is absolutely necessary

ff05 committed:
removed machine-id fix in favor of change in template

also changed variable proxmox_cloudimage_template_node in get vm config
to proxmox_vm_node if defined or the host where the task is delegated to
